File: app/src/main/java/com/maufonseca/mauriciocv/presentation/main/SearchInterfaces.java
package com.maufonseca.mauriciocv.presentation.main;

import com.maufonseca.mauriciocv.model.Snippet;

import java.util.ArrayList;

interface SearchDisplay {
  void updateResults(ArrayList<Snippet> snippets);
}

interface SearchInteraction {
  void search(String query);
}

interface SearchPresentation {
  void onSearchResults(ArrayList<Snippet> snippets);
}

File: app/src/main/kotlin/com/maufonseca/mauriciocv/presentation/main/SearchInterfaces.kt
package com.maufonseca.mauriciocv.presentation.main

import com.maufonseca.mauriciocv.model.Snippet

import java.util.ArrayList

interface SearchDisplay {
  fun updateResults(snippets: ArrayList<Snippet>)
}

interface SearchInteraction {
  fun search(query: String)
}

interface SearchPresentation {
  fun onSearchResults(snippets: ArrayList<Snippet>)
}


